views/js/runner/runnerComponent.js

```diff
@@ -0,0 +1,249 @@
+    'lodash',
+    'context',
+    'core/promise',
+    'core/pluginLoader',
+    'core/providerLoader',
+    'ui/component',
+    'taoTests/runner/runner',
+    'tpl!taoTests/template/runnerComponent'
+], function ($, _, context, Promise, pluginLoaderFactory, providerLoaderFactory, component, runnerFactory, runnerComponentTpl) {
+    'use strict';
+
+    /**
+     * List of options required by the runner
+     * @type {String[]}
+     */
+    var requiredOptions = [
+        'provider'
+    ];
+
+    /**
+     * Some defaults options
+     * @type {Object}
+     */
+    var defaults = {};
+
+    /**
+     * Loads the modules dynamically
+     * @param {Function} loader - the loader factory
+     * @param {Object[]} modules - the collection of modules to load
+     * @returns {Promise} resolves with the list of loaded modules
+     */
+    function loadModules(loader, modules) {
+        return loader()
+            .addList(modules)
+            .load(context.bundle);
+    }
+
+    /**
+     * Registers a list of loaded providers
+     * @param providers
+     */
+    function registerProviders(providers) {
+        _.forEach(providers, function (provider) {
+            runnerFactory.registerProvider(provider.name, provider);
+        });
+        return providers;
+    }
+
+    /**
+     * Wraps a test runner into a component
+     * @param {Object}   config - The testRunner options
+     * @param {String}   config.provider - The provider to use
+     * @param {Object[]} [config.plugins] - A collection of plugins to load
+     * @param {Object[]} [config.providers] - A collection of providers to load
+     * @param {jQuery|HTMLElement|String} [config.renderTo] - An optional container in which renders the component
+     * @param {Boolean} [config.replace] - When the component is appended to its container, clears the place before
+     * @param {Number|String} [config.width] - The width in pixels, or 'auto' to use the container's width
+     * @param {Number|String} [config.height] - The height in pixels, or 'auto' to use the container's height
+     * @param {Function} [template] - an optional template for the component
+     * @returns {runnerComponent}
+     */
+    return function runnerComponentFactory(config, template) {
+        var runner, resolver;
+
+        // the runner promise is built aside for architecture reason
+        // otherwise the errors that may be raised from the init() method wouldn't be properly managed
+        var runnerPromise = new Promise(function (resolve, reject) {
+            resolver = {
+                resolve: resolve,
+                reject: reject
+            };
+        });
+
+        var componentTemplate = template || runnerComponentTpl;
+
+        var runnerComponentApi = {
+            /**
+             * Does the option exists ?
+             * @param {String} name - the option key
+             * @returns {Boolean}
+             */
+            hasOption: function hasOption(name) {
```

Comment:
`Options` management should go away too. It's not necessary and too late at runtime. 
Verify the required options before instantiating the component and throw a `TypeError`.